x/kavamint/genesis.go

```diff
@@ -4,30 +4,35 @@ import (
 	"fmt"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	"github.com/kava-labs/kava/x/kavamint/keeper"
 	"github.com/kava-labs/kava/x/kavamint/types"
 )
 
 // InitGenesis new mint genesis
-func InitGenesis(ctx sdk.Context, keeper keeper.Keeper, ak types.AccountKeeper, data *types.GenesisState) {
-	keeper.SetParams(ctx, data.Params)
-
-	if data.PreviousBlockTime.IsZero() {
-		panic(fmt.Sprintf("No previous_block_time set in genesis state of %s module", types.ModuleName))
+func InitGenesis(ctx sdk.Context, keeper keeper.Keeper, ak types.AccountKeeper, gs *types.GenesisState) {
+	// guard against invalid genesis
+	if err := gs.Validate(); err != nil {
+		panic(fmt.Sprintf("failed to validate %s genesis state: %s", types.ModuleName, err))
 	}
-	keeper.SetPreviousBlockTime(ctx, data.PreviousBlockTime)
 
-	if macc := ak.GetModuleAccount(ctx, types.ModuleName); macc == nil {
+	// get module account -- creates one with allowed permissions if it does not exist
+	macc := ak.GetModuleAccount(ctx, types.ModuleName)
```

Comment:
This actually creates the module account, so the panic is just a double check incase that behavior changes

Reply:
I wrote it both ways, but thought this was cleaner and easier to read without the permission check nested.

Reply:
Could go either way on it, basically just increased the scope of `macc` to use below